src/librustc_mir/util/liveness.rs

```diff
@@ -236,25 +270,40 @@ impl<'tcx> Visitor<'tcx> for DefsUses {
 
             LvalueContext::Inspect |
             LvalueContext::Consume |
-            LvalueContext::Validate |
+            LvalueContext::Validate => {
+                if self.mode.include_regular_use {
+                    self.defs_uses.add_use(local);
+                }
+            }
+
+            ///////////////////////////////////////////////////////////////////////////
+            // DROP USES
 
             // We consider drops to always be uses of locals.
             // Drop eloboration should be run before this analysis otherwise
```

Comment:
This note about when drop "eloboration" should run (which came from cbdb18650e383fd49beb442ce02d006d0daf865c) ... it applied to @Zoxc's move analysis, but does it apply to our uses here for NLL? I ask because I am pretty sure we want borrowck to run *before* drop elaboration ...?

Reply:
The comment is really kind of outdated. We do *not* want to run drop elaboration before the check -- and we are ok with drops being considered uses.